src/cmd/policy.go

```diff
@@ -21,6 +24,63 @@ type regoInput struct {
 	Files []string `json:"files"`
 }
 
+type gitlabRepoMetadata struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type githubRepoMetadata struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type genericStruct interface {
+	gitlabRepoMetadata | githubRepoMetadata | opslevel.Level
+}
+
+func structToValue[S genericStruct](s S) ast.Value {
+	marshalData, err := json.Marshal(s)
+	cobra.CheckErr(err)
+	reader := strings.NewReader(string(marshalData))
+	v, err := ast.ValueFromReader(reader)
+	cobra.CheckErr(err)
+	return v
+}
+
+func getMaxFromMap(m map[string]float64) string {
+	max := float64(0)
+	var output string
+	for k, v := range m {
+		if v > max {
+			max = v
+			output = k
+		}
+	}
+	return output
+}
+
+func convertToPercentage(m map[string]float64) map[string]float64 {
+	total := float64(0)
+	output := make(map[string]float64)
+	for _, v := range m {
+		total += v
+	}
+
+	for k, v := range m {
+		output[k] = roundFloat((v/total)*100, 1)
+	}
+	return output
+}
+
+func roundFloat(val float64, precision uint) float64 {
```

Comment:
Found this with a quick search for getting sig figs down to 1 to match what it shows in the UI.  Not sure if this is necessary.

Reply:
wow surprised this isn't built into the standard lib